cpp/src/arrow/dataset/file_base.h

```diff
@@ -404,6 +404,10 @@ struct ARROW_DS_EXPORT FileSystemDatasetWriteOptions {
   /// {i} will be replaced by an auto incremented integer.
   std::string basename_template;
 
+  /// A functor which will be applied on basename_template to replace auto incremented
+  /// integer only if provided.
```

Comment:
```suggestion
  /// A functor which will be applied on an incremented counter.  The result will be inserted into
  /// the basename_template in place of {i}.
  ///
  /// This can be used, for example, to left-pad the file counter.
```

Reply:
updated, your version is much easier to digest